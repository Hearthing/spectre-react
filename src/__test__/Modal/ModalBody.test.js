import React from 'react'
import { shallow } from 'enzyme'

import { ModalBody } from '../..'

describe('ModalBody', () => {
  it('Should render default .modal-body markup', () => {
    const wrapper = shallow(<ModalBody />)

    expect(wrapper.type()).toBe('div')
    expect(wrapper.hasClass('modal-body')).toBe(true)
  })

  it('Should render children', () => {
    const wrapper = shallow(
      <ModalBody>
        <p>hello world</p>
      </ModalBody>
    )

    expect(wrapper.children().length).toBe(1)
    expect(wrapper.find('p').length).toBe(1)
  })

  it('Should pass classnames', () => {
    const wrapper = shallow(<ModalBody className='extra' />)

    expect(wrapper.hasClass('extra')).toBe(true)
    expect(wrapper.hasClass('modal-body')).toBe(true)
  })

  it('Should render as a custom element', () => {
    const wrapper = shallow(<ModalBody renderAs='main'>custom</ModalBody>)

    expect(wrapper.text()).toBe('custom')
    expect(wrapper.hasClass('modal-body')).toBe(true)
    expect(wrapper.type()).toBe('main')
  })

  it('Should match the snapshot', () => {
    const wrapper = shallow(<ModalBody />)

    expect(wrapper).toMatchSnapshot()
  })
})
