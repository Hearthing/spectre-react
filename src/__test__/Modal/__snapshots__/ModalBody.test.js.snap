// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBody Should match the snapshot 1`] = `
<div
  className="modal-body"
/>
`;
